src/Contract/Transaction.purs

```diff
@@ -415,13 +421,40 @@ balanceTxWithConstraints tx =
   in
     BalanceTx.balanceTxWithConstraints tx' ix
 
+-- | 'Throwing' variant of `balanceTxWithConstraints`. Instead of returning in
+-- | `Either`, it throws an exception on failure. Use this when errors are not
+-- | expected, and graceful recovery is either impossible or wouldn't make
+-- | sense.
+balanceTxWithConstraintsE
+  :: UnbalancedTx
+  -> BalanceTxConstraintsBuilder
+  -> Contract FinalizedTransaction
+balanceTxWithConstraintsE tx bcb = do
+  result <- balanceTxWithConstraints tx bcb
+  case result of
+    Left err -> throwError $ error $ BalanceTxError.explainBalanceTxError err
+    Right ftx -> pure ftx
+
 -- | Same as `balanceTxWithConstraints`, but uses the default balancer
 -- | constraints.
+-- |
+-- | This is a 'non-throwing' variant of this functionality. Use this when
+-- | errors are expected, and graceful recovery is possible.
 balanceTx
```

Comment:
Let's switch variants. Most users want to just get the error message, they rarely want to dispatch on it, so it's better to call the throwing variant just `balanceTx`, and non-trowing one `balanceTxE`. Same for `balanceTxWithConstraints`. Please mention the other variant in the doc-comments for each of them.

Reply:
I mostly did this to minimize breakage: this way, if people were already using `balanceTx` or `balanceTxWithConstraints`, their code would not have to change. Happy to do it the other way though: it's actually more idiomatic that way name-wise.